raiden/api/python.py

```diff
@@ -284,25 +284,26 @@ def channel_open(
             registry.get_token_network(token_address),
         )
 
-        has_enough_reserve, estimated_required_reserve = has_enough_gas_reserve(
-            raiden=self.raiden,
-            channels_to_open=1,
-        )
-
-        if not has_enough_reserve:
-            raise InsufficientGasReserve((
-                'The account balance is below the estimated amount necessary to '
-                'finish the lifecycles of all active channels. A balance of at '
-                f'least {estimated_required_reserve} wei is required.'
-            ))
-
-        try:
-            token_network.new_netting_channel(
-                partner=partner_address,
-                settle_timeout=settle_timeout,
+        with self.raiden.gas_reserve_lock:
+            has_enough_reserve, estimated_required_reserve = has_enough_gas_reserve(
+                self.raiden,
+                channels_to_open=1,
             )
-        except DuplicatedChannelError:
-            log.info('partner opened channel first')
+
+            if not has_enough_reserve:
+                raise InsufficientGasReserve((
```

Comment:
This will not work. Example (race works with any concurrent transactions):

- [estimate from `setTotalDeposit`](https://github.com/raiden-network/raiden/blob/c22a189589c66f767fdc7d992d4d1cdb1963d3bd/raiden/network/proxies/token_network.py#L605-L611) is being processed
- [has_enough_gas_reserve at channel_open](https://github.com/palango/raiden/blob/6c3d180e89e1dd4001ee7b2b2bca9fa09ff91375/raiden/api/python.py#L288) is called here
- `has_enough_gas_reserve` above completes, and says `yes`
-  `new_netting_channel` is called, and it context-switches (tons of reasons [e.g. logging](https://github.com/raiden-network/raiden/blob/c22a189589c66f767fdc7d992d4d1cdb1963d3bd/raiden/network/proxies/token_network.py#L165))
- `setTotalDeposit` is called first, breaking the assumption above

This will always be the case, unless the transaction/nonce lock in the JSON RPC is used

Reply:
I'm not sure if I understand, but you talk about a channel opening and a deposit running concurrently (with the deposit being for a different channel).

Then the deposit would be priced in for the open channel already and would not cause problems, right?